SDK Installer test on all platforms

tijcolem opened the pull request:
Pull request overview
---------------------
This adds in a new GitHub actions to run regressions tests on all platforms using both cli and ruby bindings.  I had to create a fork in order to test the GitHub actions as it needs to be on develop (at least I think it does).  It's setup pretty much the same as the Test SDK Installer except the branch commits. I suppose this could be added but I think we can use the Test SDK Installer workflow to commit the test results.  

This one is still running as of now.  https://github.com/tijcolem/OpenStudio-resources/actions/runs/1397133553 
Just waiting to make sure all steps work but I think this is close.

tijcolem committed:
test using all supported platforms in GH actions

tijcolem committed:
remove the branch name as not using it

tijcolem committed:
Add different workflow title

tijcolem committed:
Merge pull request #1 from tijcolem/expand_platforms

Expand platforms

tijcolem committed:
try and shorten the description input labels

tijcolem committed:
fix yaml

tijcolem committed:
remove extra 2004 block

tijcolem committed:
use curl vs wget on windows

tijcolem committed:
try again on windows download

tijcolem committed:
try again on windows download

tijcolem committed:
try and fix mac installer

tijcolem committed:
let's try and test using ruby bindings

tijcolem committed:
enable ruby binding test on all platforms

tijcolem committed:
add both cli and ruby bindings tests

tijcolem committed:
fix mac nproc cmd

tijcolem committed:
fix mac nproc again

jmarrec commented on ci/install-windows.qs:
```diff
@@ -0,0 +1,41 @@
+function Controller () {
```
Same as the mac one really you need a single install-script.qs

tijcolem replied:
Fair point. this is done

jmarrec commented on .github/workflows/manual_installer_all_platforms.yml:
```diff
@@ -0,0 +1,327 @@
+      os_installer_ubuntu_1804:
+        description: 'Ubuntu 1804 (.deb) URL'
+        required: true
+        default: 'https://github.com/NREL/OpenStudio/releases/download/v3.2.1/OpenStudio-3.2.1%2Bbdbdbc9da6-Ubuntu-18.04.deb'
```
This is tedious to launch. We can infer the installer links for all platforms from a single one supplied.

tijcolem replied:
I suppose OS builds will always have the base url.

tijcolem replied:
Well I take that back. Windows is a little odd as I push out both unsigned and signed installers so s3 on nightly build that we'll certainly want to test and hook into this action. 

e.g. http://openstudio-ci-builds.s3-website-us-west-2.amazonaws.com/?prefix=3.3.0-rc2/ 
signed: http://openstudio-ci-builds.s3-website-us-west-2.amazonaws.com/?prefix=3.3.0-rc2/signed/

Even though it's a bit tedious to enter 4 links there may be cases it's needed.

jmarrec replied:
So you went back on this idea of having 4 links?

tijcolem replied:
It's now just base url along with build sha, version.

jmarrec commented on .github/workflows/manual_installer_all_platforms.yml:
```diff
@@ -0,0 +1,327 @@
+        ruby -v
+        MT_CPU=$(nproc) ruby model_tests.rb
+
+    - name: Generate HTML and heatmap
```
You probably mean to run this step twice, after each run of model_tests.rb (cli + run)

You probably want to run with CUSTOMTAG=cli and CUSTOMTAG=ruby

And call process_results.py with --tagged or --all

tijcolem replied:
Yep, that's a good idea to tag them as cli and ruby

jmarrec replied:
Do we really want to run model_tests.rb with both openstudio (CLI) and system ruby? That's gonna take a long time. The fact that we would want to do it on Windows I can see, but linux/mac I'm not sure.

tijcolem replied:
I plan on these running nightly so I don't mind if they take a while to run. If we want to hook this into PRs on OpenStudio I think we should just run cli and maybe just one platform but we can use the other workflow for that. Maybe rename the workflow to <platform>_installer or something along those lines.

jmarrec commented on .github/workflows/manual_installer_all_platforms.yml:
```diff
@@ -0,0 +1,327 @@
+
+jobs:
+  installer_ubuntu_1804:
+    runs-on: ubuntu-18.04
```
There's code that's duplicated for each job, but using a matrix and a bunch of if platform (in bash or conditionally run some steps) is probably not going to simplify much so I'm fine with it

tijcolem replied:
Yeah, I thought of this. My understanding is the matrix works great for testing things such as multiples versions of python, not sure how if matrix works for vm instances.

jmarrec reviewed (changes requested):
Couple of small adjustments needed but it's 95% there! Nice!

tijcolem committed:
use single qt install script

tijcolem committed:
add tags to cli and ruby and process these using tagged option

tijcolem committed:
Merge remote-tracking branch 'nrel/develop' into develop

tijcolem committed:
make the all platform action use a single base url

tijcolem committed:
add / to end of urls

tijcolem committed:
fix description

tijcolem committed:
add proper default value

tijcolem committed:
remove end /

tijcolem committed:
fix mac file type

jmarrec commented on .github/workflows/manual_installer_all_platforms.yml:
```diff
@@ -37,9 +37,10 @@ jobs:
       run: |
         set -x
-        echo "Installer link: ${{ github.event.inputs.os_installer_ubuntu_1804 }}"
+        installer_url="${{ github.event.inputs.base_url}}OpenStudio-${{ github.event.inputs.os_version}}${{ github.event.inputs.os_prerelease_tag}}%2B${{ github.event.inputs.os_build_sha}}-Ubuntu-18.04.deb"
```
maybe a bit fragile, if you forgot the trailing slash in base_url?

jmarrec commented on .github/workflows/manual_installer_all_platforms.yml:
```diff
@@ -37,9 +37,10 @@ jobs:
       run: |
         set -x
-        echo "Installer link: ${{ github.event.inputs.os_installer_ubuntu_1804 }}"
+        installer_url="${{ github.event.inputs.base_url}}OpenStudio-${{ github.event.inputs.os_version}}${{ github.event.inputs.os_prerelease_tag}}%2B${{ github.event.inputs.os_build_sha}}-Ubuntu-18.04.deb"
```
Same would need to be applied to all jobs

```suggestion
        base_url="${{ github.event.inputs.base_url}}"
        # remove trailing slash if any
        base_url=${base_url%/}
        installer_url="$base_url/OpenStudio-${{ github.event.inputs.os_version}}${{ github.event.inputs.os_prerelease_tag}}%2B${{ github.event.inputs.os_build_sha}}-Ubuntu-18.04.deb"
```

tijcolem replied:
yeah, this would be good to add.

jmarrec commented on .github/workflows/manual_installer_all_platforms.yml:
```diff
@@ -3,22 +3,22 @@ name: Test OS SDK Installer - All Platforms
-      os_installer_ubuntu_1804:
-        description: 'Ubuntu 1804 (.deb) URL'
+      base_url:
+        description: 'The base url to use to download each platform'
```
Do we expect to use builds from s3 instead of a prerelease? just curious

tijcolem replied:
yes, I recently used s3 as a test and works fine.

jmarrec committed:
Force an Always On Supply Air Fan Operating Mode Schedule for PTAC/PTHP

cf https://github.com/NREL/OpenStudio/pull/4701
cf https://github.com/NREL/OpenStudio/issues/4699

tijcolem committed:
Merge remote-tracking branch 'remotes/nrel/develop' into develop

tijcolem committed:
updating ext for mac installer

tijcolem committed:
update required to false for prerelease tag

Commit:
update 3.4.0 tests

tijcolem committed:
Merge branch 'develop' of https://github.com/tijcolem/OpenStudio-resources into develop

tijcolem committed:
merge latest from nrel/OpenStudio-resources

tijcolem committed:
update to use 22.04 runner

tijcolem committed:
fixes for url paths and ruby action

tijcolem committed:
update mac runner to macos-11
